worlds/rabi_ribi/client/patch.py

```diff
@@ -120,8 +134,9 @@ def remove_item_from_map(ctx: RabiRibiContext, area_id: int, x: int, y: int):
     f.close()
 
 def embed_seed_player_into_mapdata(ctx: RabiRibiContext, item_modifier):
-    for area_id, _ in item_modifier.stored_datas.items():
-        f = open(f"{ctx.custom_seed_subdir}/area{area_id}.map", "r+b")
-        f.seek(MAP_TILES0_OFFSET)
-        f.write(ctx.seed_player_id.encode())
-        f.close()
+    if ctx.seed_player_id:
```

Comment:
I would advise throwing an error if we get to this point and ctx.seed_player_id is not populated. The client uses this seed to confirm the player is playing the correct AP, and will not send items back and forth if it doesn't see the correct ID.

If this fails to get populated, its probably better to throw an explicit error stating so. Otherwise this will surface as a "not on the correct Rabi Ribi AP instance" to the player, which isn't quite right.

Reply:
I originally just added these checks to clear warnings, but agreed. I think we do checks outside the patcher already, but this should help catch any breaking changes to the client.